test/Sema/rdar129359362.swift

```diff
@@ -0,0 +1,18 @@
+// RUN: %empty-directory(%t)
+// RUN: split-file %s %t
+// RUN: %swift-frontend -emit-module %t/Lib.swift -o %t/Modules/Lib.swiftmodule -enable-library-evolution -experimental-skip-all-function-bodies -experimental-lazy-typecheck
+// RUN: %swift-frontend -typecheck %t/Client.swift -I %t/Modules
+
+//--- Lib.swift
+
+public class MyClass {
+  public lazy var lazyVar: Int = { 2 }()
+}
+
+//--- Client.swift
+
+import Lib
+
+func test(foo: MyClass) {
```

Comment:
Rather than having a dedicated test-case for this, you might consider extending `Serialization/lazy-typecheck` (which is based on `/Inputs/lazy_typecheck.swift` and `/Inputs/lazy_typecheck_client.swift`). The benefit would be that it will get exercised in a number of different ways.

Reply:
Thanks, I'll do that!